### DISCUSSION TITLE
Some gratitude and a project

### DISCUSSION THREAD

**Discussion by amrojjeh** [unpoly/unpoly#579] [Show and tell]
First of all, thank you guys so much for developing Unpoly! My project used to be in HTMX and while their contributions have been great, I've noticed their API is awkward and hard to memorize, so I looked at other options. I found Turbo which solved all my issues with HTMX (no more inner/outer), but I heard the docs were lacking, so that's when I checked out Unpoly.

Going from HTMX to Unpoly was _extremely_ easy. It took about roughly 6 or so minutes. On the same day, I started experimenting with transitions and a few other features, but I decided to roll them back since the way the UI is designed, it would require a bit of polish on my end.

However, two features really helped this project go: `up.compiler` and `link.preload`. Before I integrated HTMX, the app was done as a prototype completely in the frontend, meaning all the resources would be loaded when the site was fetched, making all the interactions extremely fast. Of course, that's not really practical once the site grows, but after I integrated a server and migrated all the frontend code to an HTMX style backend, I missed the speed in which I can cycle through the multiple choice, and HTMX doesn't provide an easy way to run Hotwired style connectors. I realized though that if I could just cache the four choices ahead of time using `preload` and `compiler`, I can get the same speed as the total front end version without changing the backend at all, and that's what I did, and I was pleasantly surprised. It was a night and day difference.

Anyhow, I just wanted to talk about how great Unpoly has been and how some of its features really affected the shape of the program. If you want to try out yourself, here's the link: https://nahw.app/text/1/0

Note that the purpose of the app is to aid intermediate Arabic students in their practice, so it might be hard to use. If you just want to see what I'm talking about, use the shortcuts labeled between 1-4, and you'll see the speed. It's awesome and it feels like it's an SPA.

Thanks y'all!

**Comment by amrojjeh** [comment]
I just realized that the library is mostly developed by @triskweline , so thanks!

**Reply by triskweline** [reply]
I'm happy that you found some success with Unpoly. Thanks for taking the time to write this, and good luck with your project!
